### ISSUE TITLE
===========

### ISSUE THREAD

**Issue by Yannaing33** [elementor/elementor#23896]
===========
# [Beta 6 UPDATED - PLEASE READ BELOW!](https://github.com/elementor/elementor/discussions/19854#discussioncomment-3972239)
===========

![Cover for Hub](https://user-images.githubusercontent.com/76733085/191049686-426bf5c3-5128-42a5-bf0f-2b1f6e01f21d.png)

**Hi Elementor Beta Testers** 👋

We are incredibly excited to introduce our new beta versions of Elementor 3.8 and Elementor Pro 3.8, which includes some highly anticipated features. We also take this opportunity to introduce to you our new release mechanism, which will make our future releases a lot more predictable, and reliable.

Try out the free and Pro Beta versions of 3.8 and let us know what you think about our newest additions, including:

- Loop Builder (Pro)
- Flexbox Container - Marked as Stable
- Remove Kit - Revert kits to their previous condition

# Introducing The Loop Builder - Pro feature

![01 Loop Builder (Pro)](https://user-images.githubusercontent.com/76733085/191049844-66de623d-14b7-460b-8c94-fb26bd9c85f0.png)

Elementor 3.8 will be introducing the first phase of the Loop Builder. The Loop Builder enables you to create and design the main item for repeating content (i.e. loops) such as listings, as well as customize its layout on the page. By creating a main item template yourself, you gain a lot more design and dynamic content flexibility than the Posts widget allows you to achieve.

The first phase of the Loop Builder includes the ability to create a main item for posts, and custom post types, future updates will include full WooCommerce support, and additional capabilities that will enhance your website.

### Creating your Main Item Template 

There are two ways to design your main item template:
- In-Place Editing - available by dragging the Loop Grid widget onto a page and clicking the ‘Create a Template’ button.
- Blank Canvas - available by adding a new Loop template from the Theme Builder, or by adding a new template from the Templates menu in the WordPress dashboard.

To design your main item template, you can use nearly any of the widgets in Elementor. Each of the widgets you use, can be assigned any of the dynamic values available on your website, or they can be left as static and used to enhance the design of your main item.

### Displaying the Loop on a Page 
The loop templates you create can be used on any page, using the Loop Grid widget. With this widget, you can determine the name of the template you would like to use, the data source from WordPress to display content from, and adjust the layout, style and pagination to your liking.

- Since the Loop is dependent on Flexbox Containers, to use the Loop Builder on your website, you will need to enable the Flexbox Containers experiment.
- After activating the Flexbox Containers experiment, you will also need to activate the Loop experiment.

#### Important Note for Cloud Hosted Website Users 
Please note that in order to eliminate conflict with other third party plugins, the naming convention for the Loop template has changed from ‘Loop’ to ‘Loop Item.’ As a result, if you’ve created loop templates with previous Cloud versions of Elementor, they will be deleted. We apologize for any inconvenience this may have caused. On the new version, this will be solved and templates will be saved properly.

#### How to Test It:
**Create an In-place loop**

1. Make sure you have at least six posts on your website filled with the post title, featured images, and any additional content you want to use.
2. Create a new page, edit it with Elementor and drag in the Loop Grid widget.
3. Click on the Create Template button on the panel or in the Editor’s edit mode.
4. Ensure that you are switching to an In-Place editing mode, within the main item’s container, and that the widgets panel suggests recommended Posts related widgets first.
5. Drag and Drop any dynamic (and static) widget into the template area.
6. Once the main item design is ready, click on the “Save & Back” yellow handle.
7. Make sure you switch back to the parent page.
8. Ensure the other posts in the loop are populated automatically based on the main template design.
9. Click on the “Edit Template” button or handle and switch to the main template design.
10. Change the main item design, save, and return to the parent page.
11. Ensure again that the design of the other posts in the loop change based on the new design of the main item.
12. Go over the layout controls and change the Columns, Items Per Page and switch on Masonry.
13. Check the Query section and make sure all the controls work as expected.
14. Go over the Pagination controls and test the different options.
15. Go over to the Style tab and start examining the layout and pagination possibilities.
16. Make sure everything works as expected in the Editor and on the Frontend

**Create a loop item from the Theme Builder**

1. Go to the Theme Builder (New UI and WordPress Table view), and create a new loop item template by clicking on the “Add New” Button
2. Ensure you can see the recommended Posts related widgets.
3. Drag and Drop any dynamic (and static) widget  into the template area.
4. Adjust the Preview width by clicking on the template settings icon (cogwheel icon). Then click on the Preview Settings section, and play with the width slider control.
5. Rename the template name in the General Settings section (from the cogwheel icon).
6. Once the main item design is ready, click on Publish.
7. Create a new page, edit it with Elementor and drag in the Loop Grid widget
8. Select the new template under the “Choose a template” option.
9. Ensure the loop is presented correctly in the edit mode, and that the other posts in the loop are populated automatically based on the main template’s design.
10. Make sure everything works as expected in the Editor and on the Frontend

# Flexbox Containers beta - Now Marked as Stable

![02 Flexbox Containers - Marked as Stable](https://user-images.githubusercontent.com/76733085/191049902-05f3d2e4-8506-4f33-9845-6af6bf0d23b2.png)

After having gone through extensive testing, and reviewed under a variety of use cases and integrations, this version marks the status of the Flexbox Containers experiment as beta, and is now considered stable. That means you can feel confident using them on your own website.

We will make an effort to ensure that any changes and improvements made to Containers from this point forward will also be backwards compatible, and should not affect the way they appear or function on existing websites who have activated the experiment. Since for some of you, Flexbox Containers will require some learning and getting used to, for the time being - it will be your decision when to activate the experiment and use Containers on your website. [That is also why, they will still be located under the Experiment tab in Elementor Settings for the time being].

#### Please note, if you decide to activate the experiment, and design layouts using the Container, but then decide to deactivate the experiments, all website parts created with Containers may disappear.

To learn more about how you can use Flexbox Containers on your website, you can also try it out yourself first in the [Elementor Flexbox Playground](https://playground.elementor.com/demo/flexbox/)

### Please note - Markup changes in Container.
We will introduce in later Beta versions markup changes to the Container class names as described below:

-- Replace `e-container` to `e-con`
-- Replace `e-container--width-boxed` to `e-con-boxed`
-- Replace `e-container--width-full` to `e-con-full`
-- Replace `e-container__inner` to `e-con-inner`

# New: Remove Kit - Revert your Website to its Previous Condition

![03 Revert Kit - Restore kits to their previous condition](https://user-images.githubusercontent.com/76733085/191049924-91fbcb9b-f54d-4ccb-94da-7f2c735a0ccf.png)

Ever downloaded a website kit but then changed your mind? With this update, you don’t need to restore your entire website to a previous version or manually undo all the global styles applied by the kit or delete the pages it came with.

Any kit downloaded from this version onwards can be easily removed by going to the Import/Export tab under Elementor Settings → Tools section(from the WordPress dashboard) and clicking the ‘Remove Kit’ button. This button will only appear if you downloaded a kit after updating to this version.

#### How to Test It:

1. Download a website kit by going to WordPress dashboard → Templates → Kit Library 
2. See that everything works properly on the front-end and that it downloaded all the content associated with the kit
3. Try adding a new page, and see how when you create a new page, the kit’s global styles apply on the new page
4. Revert your kit to restore your website to its previous condition by going to Elementor → Tools → Import/Export Kit and click the new button at the bottom of the screen ‘Remove Kit’, then click ‘Yes’ on the action popup
5. Review your posts and pages and make sure the pages that were added with the kit have been removed
6. If you’ve created a new page, review its content to see that the content remind, and the Global styles of the page have restored to their previous condition, before downloading the kit

# Update about the Release Mechanism

Over the past couple of months, we have been reviewing ways to increase the speed of providing you with new updates, and reducing incompatibility issues.

As a result, we decided to change our release process in two ways:

- Major versions of Elementor’s Free and Pro plugin will be updated at the same time, as evident in this upcoming version release.
- Implementing a consistent release schedule:
- Consistent small releases to Cloud Hosted Websites
   - deploying to these websites first allows us to increase release frequency in a controlled environment, which results in better and more compatible version releases.

----
# More Improvements

#### Free version

* New: Revert your Website to its Previous Condition - Allow removing the last imported Kit
* New: The container experiment is now officially set as beta
* Tweak: Increase inputs in Replace URL tool to support long URLs (#19559)
* Tweak: Added Dynamic Tags for Global Colors (#15135)
* Tweak: Improved performance of Inline Fonts Icons experiment (#19447)
* Tweak: Improve browser responsiveness during Elementor's initialization process (#15228, props @FlyingDR)
* Tweak: Added Responsive control to Text Align in Columns and Sections (#13199, props @rodolphebertozzo)
* Tweak: Added size units to Letter Spacing in Typography modal (#19726)
* Tweak: Added more size units to Size and Spacing controls Icon and Icon box widgets (#19496)
* Tweak: Added clarification for the `a` tag to the container element (props @rodolphebertozzo)
* Tweak: Added text stroke to number in Counter widget (props @rodolphebertozzo)
* Tweak: Added text stroke to title in Counter widget (props @rodolphebertozzo)
* Tweak: Added text stroke to text in Divider widget (props @rodolphebertozzo)
* Tweak: Added em unit for border radius to layout elements (props @rodolphebertozzo)
* Tweak: Added text stroke to title in Image Box widget (props @rodolphebertozzo)
* Tweak: Added more size units to Spacing and Image Size in Testimonial and Image Box widgets (props @rodolphebertozzo)
* Tweak: Custom links in menu items should be relative to the site address when importing a Kit
* Tweak: Import/Export CLI and UI mechanisms were merged into a unified service
* Tweak: Improve the UX of dependencies between experiments
* Tweak: Changed Full-width and Boxed content width functionality in Container
* Tweak: Added labels to font weight numeric values (#18761) 
* Tweak: Change default Google fonts loading method to "Swap" on new sites
* Tweak: Changed default direction for Featured sorting to ASC in Kit Library
* Tweak: Added migrate script to handle retro PHP8 type error on image custom size
* Tweak: Re-organized SCSS files for the Container
* Tweak: Changed default Content Width when adding a new Container from right click context-menu
* Fix: Importer WordPress root write permissions check causes import failures (#17255)
* Fix: Insert library button is clickable while inserting a template
* Fix: Go pro link UI glitch in export kit tool
* Fix: Export kit doesn't work in a Multisite Network
* Fix: WooCommerce products and categories are not imported as expected when assigned to a menu
* Fix: CSS minified files not generated on build

#### Pro version

* New: Introducing Loop Builder feature as a beta status experiment - Create powerful & repeating loop templates and populate each one with dynamic content and design flexibility (#4440)
* Tweak: Add `wp_body_open()` to header in Header template (#11562, #15828, #13089)
* Tweak: Added support border-radius option to the Code Highlight widget (#14316)
* Tweak: Import/Export CLI and UI mechanisms were merged into a unified service
* Tweak: User with no permission to Notes cannot be mentioned in a note
* Tweak: User with no permission to view a post cannot be mentioned in a note
* Tweak: Notes was added to the right click context-menu
* Tweak: Notes panel can be resizable
* Tweak: Notes panel can be dragged outside of the canvas in responsive mode in the editor
* Tweak: Updated form validation messages translation strings in Form widget
* Tweak: Updated translators comments
* Tweak: Theme Builder logo and Title should be clickable
* Fix: Accessibility attribute `role=navigation` is redundant in Nav Menu widget (#17582)
* Fix: Note's timestamp is being updated according to the last activity in Notes (#19123)
* Fix: Share buttons is accessible with keyboard but not clickable
* Fix: Sub-items icons style is missing in preview with Inline Font Icons experiment in Nav Menu widget
* Fix: Quotes are appearing as HTML when editing a note in Notes
* Fix: Mixed Content errors on HTTPS in Video Playlist Widget (#18895, #18945)
* Fix: Label's Icon height increases when Inline Font Icons experiment is active in Hotspot Widget
* Fix: Sub conditions of templates are not overridden when importing a Kit

-----

## Make An Impact: Share Feedback Before Features Are Released
We’re always looking to get to know Elementor customers on a deeper level. If you’re interested in meeting with our team and sharing your insights, feedback, and feature requests you’re welcome to sign up for some quality time with us [here](https://elementor.com/pages/joinresearch/). Or download [Elementor’s Dev Edition](https://wordpress.org/plugins/elementor-beta/). Talk soon!

----

## How to Install the Beta Version
If you still haven't enabled beta testing in Elementor, please follow these steps:

- Go to Elementor > Tools > Versions
- Enable the beta tester feature
- Go to Dashboard > Updates and update to the latest beta version.
- This will get you set up to test the new Elementor Free and Pro 3.8 Beta versions.

----

## Known Bugs in Elementor 3.8 Free and Pro Beta

As we prepare these Beta versions for stability, there are a number of bugs we identified that are already in progress of being solved,  including:

#### Loop Builder

* Fix: Columns control is not working on responsive devices in Loop Grid widget - **Bug Solved**
* Fix: Loop typography doesn't render correctly from a template created in Theme Builder in Loop widget - **Bug Solved**
* Fix: Button widget styles inside a loop template are affected from load more pagination button styles in Loop Grid widget - **Bug Solved**
* Fix: Featured Image dynamic tag is not working as expected in loop template on a container level - **Bug Solved**
* Fix: Error message appears when clicking the discard button when creating or editing a template - **Bug Solved**
* Fix: Link styles in Theme Styles effects loop's CTA button - **Bug Solved**
* Fix: Loop template does not change live when editing the design in an in-place editing - **Bug Solved**

#### Flexbox Container

* Fix: Sticky functionality is not working as expected in Container - **Bug Solved**

----

## Notes

As we make changes to our GitHub in an effort to keep it organized, improve the way we deliver information to you, and our ability to review your feedback, please follow the instructions below when reporting bugs, or making feature requests:

- To report a bug related to beta versions - open a new [Bug report](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml)
- For feature requests - open a new [Feature request issue](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=feature_request.yml)
- For other comments specifically about this beta version - comment here

Designated time for release: `October 30, 2022 (Estimated)`

Many thanks for your support and help!

Cheers,
_Elementor Team_

_Originally posted by @avivu in https://github.com/elementor/elementor/discussions/19854_

**Comment by portrayaloflife** [comment]
Will the role manager be functional again in this release? It currently is not and were told you're making changes or fixes to it. Really hoping it can be a part of 3.8 so our clients can be able to update their pages again.

**Comment by nicholaszein** [comment]
Hello @Yannaing33,

please do not post issues by selecting to reference a comment in a new issue. Instead, use the [New Issue](https://github.com/elementor/elementor/issues/new/choose) form.

Spam will not be tolerated.

Thanks
